modules/edge-identity/src/identity.rs

```diff
@@ -118,9 +124,12 @@ decl_module! {
 		pub fn attest(origin, identity_hash: T::Hash, attestation: Attestation) -> Result {
 			let _sender = ensure_signed(origin)?;
 			ensure!(!Self::frozen_accounts().iter().any(|i| i == &_sender.clone()), "Sender account is frozen");
-
+			// Grab record
 			let record = <IdentityOf<T>>::get(&identity_hash).ok_or("Identity does not exist")?;
-
+			// Ensure the record isn't expired if it still exists
+			if let Some(time) = record.expiration_time {
+				ensure!(time >= <system::Module<T>>::block_number(), "Identity has reached expiration");
+			}
 			if record.stage == IdentityStage::Verified {
 				return Err("Already verified");
```

Comment:
Could make this an `ensure` while you're at it.

Reply:
sounds good